drive/tests/deterministic_root_hash.rs

```diff
@@ -59,6 +60,8 @@ fn test_root_hash(drive: &mut Drive, db_transaction: &OptimisticTransactionDBTra
     let app_hash = drive
         .grove
         .root_hash(Some(db_transaction))
+        .ok()
+        .flatten()
```

Comment:
This is weird, especially since rootHash can no longer be None.

Reply:
it still can be None, but now it is also wrapped in `Result`, because we no longer cache root tree and have to visit storage and possibly fail

Reply:
I mean we should probably remove the Option in root_hash call

Reply:
What to return on empty database?